bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/TableJsonDataSource.java

```diff
@@ -106,131 +101,76 @@ public Object build() {
-            ArrayValue jsonArray = new ArrayValueImpl(new BArrayType(PredefinedTypes.TYPE_JSON));
-            if (data != null) {
-                BArray dataArray = data.getArrayValue(key);
-                for (int i = 0; i < dataArray.size(); i++) {
-                    Object value = dataArray.get(i);
-                    if (value instanceof String) {
-                        jsonArray.append(value);
-                    } else if (value instanceof Boolean) {
-                        jsonArray.append(value);
-                    } else if (value instanceof Long) {
-                        jsonArray.append(value);
-                    } else if (value instanceof Double) {
-                        jsonArray.append(value);
-                    } else if (value instanceof Integer) {
-                        jsonArray.append(value);
-                    } else if (value instanceof Float) {
-                        jsonArray.append(value);
-                    } else if (value instanceof DecimalValue) {
-                        jsonArray.append(((DecimalValue) value).floatValue());
-                    }
-                }
-            }
-            return jsonArray;
-        } else {
-            MapValue<BString, Object> jsonData = new MapValueImpl<>(new BMapType(PredefinedTypes.TYPE_JSON));
-            boolean structError = true;
-            if (data != null) {
-                Type internalType = structFields[index].getFieldType();
-                if (internalType.getTag() == TypeTags.OBJECT_TYPE_TAG
-                        || internalType.getTag() == TypeTags.RECORD_TYPE_TAG) {
-                    BField[] internalStructFields =
-                            ((BStructureType) internalType).getFields().values().toArray(new BField[0]);
-                    for (int i = 0; i < internalStructFields.length; i++) {
-                        BString internalKeyName = StringUtils.fromString(internalStructFields[i].getFieldName());
-                        Object value = data.get(internalKeyName);
-                        if (value instanceof BigDecimal) {
-                            jsonData.put(internalKeyName, ((BigDecimal) value).doubleValue());
-                        } else if (value instanceof MapValueImpl) {
-                            jsonData.put(internalKeyName,
-                                         getStructData((MapValueImpl) value, internalStructFields, i, internalKeyName));
-                        } else {
-                            jsonData.put(internalKeyName, value);
-                        }
-                        structError = false;
-                    }
-                }
-            }
-            if (structError) {
-                throw new BallerinaException("error in constructing the json object from struct type data");
-            }
-            return jsonData;
+                throw new BallerinaException("cannot construct the json object from '" + type + "' type data");
```

Comment:
```suggestion
                throw new BallerinaException("cannot construct json object from '" + type + "' type data");
```
A small suggestion on the error message.